SPECS/pigz/pigz.spec

```diff
@@ -28,8 +29,41 @@ mkdir -p %{buildroot}%{_bindir}/
 install -p -m 755 pigz %{buildroot}%{_bindir}/
 install -p -m 755 unpigz %{buildroot}%{_bindir}/
 
-%post -p /sbin/ldconfig
-%postun -p /sbin/ldconfig
+%post
+/sbin/ldconfig
+boot_configuration_file="/etc/default/grub"
+system_defined_mods_hdr="# BEGIN SYSTEM-DEFINED MODIFICATIONS:"
+parameter="foo=bar"
+
+# Check that the configuration file exists
+if [ ! -f "$boot_configuration_file" ]; then
+    echo "Error: $boot_configuration_file does not exist"
+    exit 1
+fi
+
+parameter="# cmdline snippet added by %{name}-%{version}:\nGRUB_CMDLINE_LINUX=\"\$GRUB_CMDLINE_LINUX $parameter\"\n# end cmdline snippet added by %{name}-%{version}\n"
+
+# Search for the system_defined_mods_hdr string in the configuration file
+if grep -q "$system_defined_mods_hdr" "$boot_configuration_file"; then
+    # If found, use sed to replace the string with the string plus the passed parameter
+    sed -i "s/$system_defined_mods_hdr/$system_defined_mods_hdr\n$parameter/" "$boot_configuration_file"
+else
+    # If not found, append the string plus the passed parameter to the end of the file
+    echo "Error: splicing in cmdline parameter failed for $parameter in $boot_configuration_file"
+fi
+
+%postun
+/sbin/ldconfig
+boot_configuration_file="/etc/default/grub"
+header="# cmdline snippet added by %{name}-%{version}"
+
+# Check that the configuration file exists
+if [ ! -f "$boot_configuration_file" ]; then
+    echo "Error: $boot_configuration_file does not exist"
+    exit 1
```

Comment:
i think it's fine to log/echo this, but preventing rpm uninstall seems too heavy?

Reply:
Good point. What about the reverse, when we are installing and fail to locate the /etc/default/grub?